FreeRTOS/Demo/CORTEX_M3_MPS2_QEMU_GCC/main_full.c

```diff
@@ -285,21 +319,25 @@ HeapStats_t xHeapStats;
 			/* These tasks are only created when preemption is used. */
 			if( xAreTimerDemoTasksStillRunning( xCycleFrequency ) != pdTRUE )
 			{
+				ullTimerDemoTaskFailures++;
 				pcStatusMessage = "Error: TimerDemo";
 			}
 		}
 		#endif
 
 		if( xAreStreamBufferTasksStillRunning() != pdTRUE )
```

Comment:
Added this into the conversation - should have added here:

This is a good idea, but I don't think it will work for a couple of reasons. First errors discovered in tests are latched - so once a test reports an error it will continue to do so every time the test is re-checked. Second the check task uses "else if" for each check, so only the first failure on each will result in the counter getting incremented.

You may also like to check "main_full.c" in this project to ensure the GCC project is running all the same tests: #503

Reply:
I will go through that link and check. I think it would still be valuable to keep counting, even though it is latched, so we can approximate how early it fell over? I'll change the "else if" statements to if, and create a separate boolean to see how many cycles it was running successfully for.